src/boot/bootctl.c

```diff
@@ -780,10 +809,17 @@ static int install_binaries(const char *esp_path, bool force) {
         if (!d)
                 return log_error_errno(errno, "Failed to open \""BOOTLIBDIR"\": %m");
 
+        const char *suffix = ".efi", *suffix_signed = ".efi.signed";
+        if (!streq_ptr(arg_arch, "all")) {
+                const char *arch = arg_arch ?: efi_arch();
+                suffix = strjoina(arch, suffix);
+                suffix_signed = strjoina(arch, suffix_signed);
+        }
```

Comment:
btw, i think we should generate an error if the selected arch is not supported, i.e. no binary installed for this arch

Reply:
This should probably be done separately, as it currently doesn't warn if nothing was copied.